src/main/java/com/example/jpa_relacje_26_zadaniekoncowe/recipe/RecipeController.java

```diff
@@ -0,0 +1,89 @@
+package com.example.jpa_relacje_26_zadaniekoncowe.recipe;
+
+import com.example.jpa_relacje_26_zadaniekoncowe.category.Category;
+import com.example.jpa_relacje_26_zadaniekoncowe.category.CategoryRepository;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.*;
+
+@Controller
+public class RecipeController {
+
+    private final RecipeRepository recipeRepository;
+    private final CategoryRepository categoryRepository;
+
+    public RecipeController(RecipeRepository recipeRepository, CategoryRepository categoryRepository) {
+        this.recipeRepository = recipeRepository;
+        this.categoryRepository = categoryRepository;
+    }
+
+    @GetMapping("/recipes")
+    public String recipes(Model model, @RequestParam(required = false) String categoryName) {
+
+        Iterable<Recipe> recipeList;
+
+        if (categoryName != null) {
+            recipeList = recipeRepository.findRecipeByCategoryName(categoryName);
+            model.addAttribute("category", categoryRepository.findByName(categoryName));
+        } else {
+            recipeList = recipeRepository.findAll();
+        }
+        model.addAttribute("recipesList", recipeList);
+        return "recipesPage";
+    }
+
+    @GetMapping("/recipe/{id}")
+    public String recipe(Model model, @PathVariable Long id) {
+        Optional<Recipe> recipeOpt = recipeRepository.findById(id);
+        if (recipeOpt.isPresent()) {
+            model.addAttribute("selectedRecipe", recipeOpt.get());
+            return "oneRecipePage";
+        } else {
+            return "error";
+        }
+    }
+
+    @GetMapping("/add-like/{id}")
+    public String addLikeToRecipe(@PathVariable Long id) {
+        recipeRepository.addLikeForRecipe(id);
+        return "redirect:/recipe/" + id;
+    }
+
+    @GetMapping("/add-new-recipe")
+    public String addForm(Model model, @RequestParam(required = false) Long recipeid) {
+        if (recipeid != null) {
+            Optional<Recipe> recipeOpt = recipeRepository.findById(recipeid);
+            recipeOpt.ifPresent(recipe -> model.addAttribute("newRecipe", recipe));
+        } else {
+            model.addAttribute("newRecipe", new Recipe());
+        }
+        model.addAttribute("categoriesList", categoryRepository.findAll());
+        return "addRecipePage";
+    }
```

Comment:
Wykorzystanie tego samego formularza do dodawania i edycji to dobry pomysł, ale rozbij to na dwie metody z różnym mapowaniem /add-new-recipe i /edit żeby nie wprowadzać w błąd